src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/DynamicInvokeInfo.cs

```diff
@@ -636,6 +638,14 @@ private void ThrowForArgCountMismatch()
                     EETypePtr srcEEType = arg.GetEETypePtr();
                     EETypePtr dstEEType = argumentInfo.Type;
 
+                    // Destination type can be null if we don't have a MethodTable for this type. This means one cannot
+                    // possibly pass a valid non-null object instance here.
+                    if (dstEEType.IsNull)
+                    {
+                        Debug.Assert(i != 0 || _isStatic);
+                        throw InvokeUtils.CreateChangeTypeArgumentException(srcEEType, Method.GetParametersAsSpan()[i - (_isStatic ? 0 : 1)].ParameterType, destinationIsByRef: false);
+                    }
+
                     if (!(srcEEType.RawValue == dstEEType.RawValue ||
```

Comment:
I would move the null check to be after the exact type match. Something like:

if (srcEEType.RawValue != dstEEType.RawValue)
{
    if (dstEEType.IsNull)
    {
        ...
    }

    if (!(RuntimeImports.AreTypesAssignable(srcEEType, dstEEType) ...

}